### PR TITLE
Switch to using stable values for Threads

### PR DESCRIPTION
**Pull request by gsouquet**
Related to https://github.com/vector-im/element-web/issues/21215

- Add check for stable/unstable thread server support
- Use NamespacedValue for threads prefixes
- determine server support before first sync
- remove checks in server support has already been checked

---
This PR currently has no changelog labels, so will not be included in changelogs.

Add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is plus `X-Breaking-Change` if it's a breaking change.

### CODE REVIEW

File: src/models/thread.ts [hunk 1]
```diff
@@ -186,7 +185,7 @@ export class Thread extends TypedEventEmitter<EmittedEvents, EventHandlerMap> {
             this._currentUserParticipated = true;
         }
 
-        const isThreadReply = event.getRelation()?.rel_type === RelationType.Thread;
+        const isThreadReply = event.getRelation()?.rel_type === THREAD_RELATION_TYPE.name;
```
Comment (t3chguy) [inline src/models/thread.ts:188]: Shouldn't this use `THREAD_RELATION_TYPE.matches(relation?.rel_type)` to support both legacy and new?

File: src/models/event.ts [hunk 2]
```diff
@@ -505,7 +505,7 @@ export class MatrixEvent extends TypedEventEmitter<EmittedEvents, MatrixEventHan
      */
     public get threadRootId(): string | undefined {
         const relatesTo = this.getWireContent()?.["m.relates_to"];
-        if (relatesTo?.rel_type === RelationType.Thread) {
+        if (relatesTo?.rel_type === THREAD_RELATION_TYPE.name) {
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/models/event.ts:508]: I think using NamespacedValue for this then will be confusing, as that implies both will exist at once.

[the remark this answers is not in the document]
Comment (gsouquet) [reply src/models/event.ts:508]: What do you recommend to use then? That's the best middleground I could find, as I don't think there's a precedent in the codebase to do what we're trying to achieve here

[the remark this answers is not in the document]
Comment (t3chguy) [reply src/models/event.ts:508]: I think this would be a new class, or just a getter on Thread or something

File: src/NamespacedValue.ts [hunk 3]
```diff
@@ -70,6 +70,22 @@ export class NamespacedValue<S extends string, U extends string> {
...
+    public setPreferUnstable(preferUnstable: boolean): void {
+        this.preferUnstable = preferUnstable;
+    }
+
+    public get name(): U | S {
+        if (this.stable && !this.preferUnstable) {
+            return this.stable;
+        }
+        return this.unstable;
+    }
```
Comment (t3chguy) [inline src/NamespacedValue.ts:86]: maybe we should also override `.altName` to return the same as `.name`

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

**Pull request merged**